astropy/io/fits/fitstime.py

```diff
@@ -241,21 +329,37 @@ def _convert_time_column(col, column_info):
         The time coordinate column to be converted to Time.
     column_info : dict
         Column-specific time reference frame override information.
+    global_info : dict
+        Global time reference frame information.
     """
 
     # The code might fail while attempting to read FITS files not written by astropy.
 
-    # Read in time coordinate column as Time
-    if col.shape[-1] == 2 and col.ndim > 1:
-        col = Time(col[..., 0], col[..., 1], format='jd',
-                   scale=column_info['scale'],
-                   location=column_info['location'])
-    else:
-        warnings.warn(
-            'Time column {} is not in the astropy required (jd1, jd2) format. '
-            'Hence, it will not be read as astropy Time'.format(col.info.name),
-            AstropyUserWarning)
-    return col
+    try:
+        # ISO-8601 is the only string representation of time in FITS
+        if col.info.dtype.kind == 'S':
```

Comment:
Could python3 failures be due to this test? Maybe check in a failed run whether dtype was `U` instead of `S`? I think this is the case since in `appveyor` the error message states:
```
E       AssertionError: assert False
E        +  where False = isinstance(<Column name='datetime' dtype='str29' length=2>\n1999-01-01T00:00:00.123456789\n          2010-01-01T00:00:00, Time)
astropy\io\fits\tests\test_fitstime.py:291: AssertionError
```
i.e., `dtype=str29`, which corresponds to numpy dtype `U29`.

(Obviously, this should be easy to check by running your tests locally in python3 - you should be *on* python3!!)